Python/compile.c

```diff
@@ -2546,13 +2576,24 @@ compiler_if(struct compiler *c, stmt_ty s)
         return 0;
 
     constant = expr_constant(s->v.If.test);
-    /* constant = 0: "if 0" Leave the optimizations to
-     * the pephole optimizer to check for syntax errors
-     * in the block.
+    /* constant = 0: "if 0"
      * constant = 1: "if 1", "if 2", ...
      * constant = -1: rest */
-    if (constant == 1) {
+    if (constant == 0) {
+        if (s->v.If.orelse) {
```

Comment:
So
```python
if 0:
    return
else:
    x = 1
```
is not error?

Please add tests for this case.

Reply:
Done